### PR TITLE
fix: shutdown error while closing replica

### PR DESCRIPTION
**Pull request by hycdong**
https://github.com/apache/incubator-pegasus/issues/906
This pull request should be reviewed with https://github.com/apache/incubator-pegasus/pull/909 together.

### CODE REVIEW

File: src/replica/replica.cpp [hunk 1]
```diff
@@ -434,10 +434,6 @@ void replica::close()
         _checkpoint_timer = nullptr;
     }
 
-    if (_app != nullptr) {
```
Comment (levy5307) [inline src/replica/replica.cpp:437]: It makes me confused that why did you remove these lines?

**Review by levy5307 — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]
